### PR TITLE
Use `layout:` for simpler configuration

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -8,6 +8,22 @@ confinement: strict
 grade: stable
 base: core18
 
+environment:
+  LC_ALL: C.UTF-8
```
Comment (Saviq) [inline snap/snapcraft.yaml]: Why this?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -17,31 +33,49 @@ apps:
...
+      - -usr/bin/miral-shell
+      - -usr/bin/miral-xrun
 
   icons:
     plugin: nil
     stage-packages: [dmz-cursor-theme]
-    organize:
-      usr/share/icons/DMZ-White: icons/default
-    prime:
-     - icons/default
```
Comment (Saviq) [inline snap/snapcraft.yaml:66]: Is this now dealt with some other way?

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -17,31 +33,49 @@ apps:
...
     plugs:
       - opengl
-    environment:
-      XKB_CONFIG_ROOT: $SNAP/usr/share/X11/xkb
-      XDG_CONFIG_HOME: $SNAP_DATA
-      XCURSOR_PATH:    $SNAP/icons
-      __EGL_VENDOR_LIBRARY_DIRS: $SNAP/etc/glvnd/egl_vendor.d:$SNAP/usr/share/glvnd/egl_vendor.d
 
 parts:
+  ppa-mir-release:
```
Comment (Saviq) [inline snap/snapcraft.yaml]: I believe you called this `ppa-mir` elsewhere?

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -17,31 +33,49 @@ apps:
...
+    plugin: nil
+    override-pull: |
+      sudo apt --assume-yes install software-properties-common
+      sudo add-apt-repository -yu ppa:mir-team/release
```
Comment (Saviq) [inline snap/snapcraft.yaml:37]: I'm quite uneasy about this approach... There's no guarantee this even works, if `snapcraft` decides to cache all dependencies prior to any of the `pull` steps.

IIUC this "solves" building outside of Launchpad builders. But they are our main consumer of this, and we can instead document that one of the Mir PPAs is required for this snap to build.

**Review by Saviq — COMMENTED** [review]
Is there a `snapcraft` issue pertaining to the PPA problem?

Also, there are conflicts to solve here.

File: snap/snapcraft.yaml (hunk 3 above)
Comment (AlanGriffiths) [reply snap/snapcraft.yaml]: Oops! I didn't notice this needs updating.

File: snap/snapcraft.yaml (hunk 4 above)
Comment (AlanGriffiths) [reply snap/snapcraft.yaml:37]: This approach has been documented multiple times on the snapcraft forums. Guaranteed or not, there are a lot of snaps that depend on it.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (AlanGriffiths) [reply snap/snapcraft.yaml]: I don't remember. Let's try without.

File: snap/snapcraft.yaml (hunk 2 above)
Comment (AlanGriffiths) [reply snap/snapcraft.yaml:66]: Yes: `bind: $SNAP/usr/share`

File: snap/snapcraft.yaml (hunk 4 above)
Comment (Saviq) [reply snap/snapcraft.yaml:37]: FWIW [snapcraft local plugins](https://docs.snapcraft.io/writing-local-plugins/5125) would be a better approach here, but as we're thinking we'd rather have a `miral-framework` instead, I'm OK with this for now.

File: snap/snapcraft.yaml [hunk 5]
```diff
@@ -5,6 +5,21 @@ description: A minimal Mir based shell for kiosk type applications
...
 
+environment:
+  #  # XDG config
+  XDG_CONFIG_HOME: $SNAP_DATA
+  XDG_RUNTIME_DIR: /run/user/0 # hardcoding as root for now
+  # Prep for Mir
+  MIR_CLIENT_PLATFORM_PATH: ${SNAP}/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/mir/client-platform
+  MIR_SERVER_PLATFORM_PATH: ${SNAP}/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/mir/server-platform
+
+passthrough:
```
Comment (Saviq) [inline snap/snapcraft.yaml]: `passthrough` no longer required.

**Review by Saviq — APPROVED** [review]
Yup.

**Pull request merged**